lib/ZumoOta/src/Zumo32U4Specifications.h

```diff
@@ -0,0 +1,165 @@
+/*
+BSD 3-Clause License
+
+Copyright (c) 2021, NewTec GmbH
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice, this
+   list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+   contributors may be used to endorse or promote products derived from
+   this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+ * @file Zumo32U4Specification.h
+ * @author Luis Moser
+ * @brief Zumo32U4Specification header
+ * @date 08/23/2021
+ * @addtogroup Zumo
+ * @{
+ */
+
+#ifndef __ZUMO32U4Specification_H__
+#define __ZUMO32U4Specification_H__
+
+#include <Arduino.h>
+
+ /** Simple struct for specifying Zumo commands */
+template <int size>
+struct ZumoCommand
+{
+    /** The binary buffer for the command data */
+    uint8_t commandData[size];
+
+    /** Size of command data buffer in bytes */
+    uint8_t commandSize;
+};
+
+/** Simple struct for specifying Zumo data */
+template <int size>
+struct ZumoData
+{
+    /** The binary buffer for the response Zumo data */
+    uint8_t data[size];
+
+    /** Size of the response data in bytes */
+    uint8_t dataSize;
+};
+
+/** Namespace for specifying all used Zumo32U4 commands, return codes and expected return values */
+namespace Zumo32U4Specification

```

Comment:
these are definitions not declarations.  It should not be in a header as every consumer will get a copy.
If is acceptable if only included in one cpp file, but should be moved out of the header in a later patch.